plugins/chat/spec/system/transcript_spec.rb

```diff
@@ -0,0 +1,246 @@
+  fab!(:chat_message_4) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:chat_message_5) { Fabricate(:chat_message, chat_channel: chat_channel_1) }
+  fab!(:topic) { Fabricate(:with_posts_topic) }
+  let(:last_topic_post) { topic.posts.last }
+
+  let(:chat_page) { PageObjects::Pages::Chat.new }
+  let(:chat_channel_page) { PageObjects::Pages::ChatChannel.new }
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+
+  before do
+    chat_system_bootstrap(current_user, [chat_channel_1])
+    sign_in(current_user)
+  end
+
+  def select_message(message, mobile: false)
+    if page.has_css?(".chat-message-container.selecting-messages")
+      chat_channel_page.message_by_id(message.id).find(".chat-message-selector").click
+    else
+      # we long press instead of hover on mobile
+      if mobile
+        chat_channel_page.message_by_id(message.id).click(delay: 0.5)
+      else
+        chat_channel_page.message_by_id(message.id).hover
+      end
+
+      # we also have a different actions menu on mobile
+      if mobile
+        find(".chat-message-action-item[data-id=\"selectMessage\"]").click
+      else
+        expect(page).to have_css(".chat-message-actions .more-buttons")
+        find(".chat-message-actions .more-buttons").click
+        find(".select-kit-row[data-value=\"selectMessage\"]").click
+      end
+    end
+  end
+
+  def cdp_allow_clipboard_access!
+    cdp_params = {
+      origin: page.server_url,
+      permission: {
+        name: "clipboard-read",
+      },
+      setting: "granted",
+    }
+    page.driver.browser.execute_cdp("Browser.setPermission", **cdp_params)
+
+    cdp_params = {
+      origin: page.server_url,
+      permission: {
+        name: "clipboard-write",
+      },
+      setting: "granted",
+    }
+    page.driver.browser.execute_cdp("Browser.setPermission", **cdp_params)
+  end
+
+  def read_clipboard
+    page.evaluate_async_script("navigator.clipboard.readText().then((clipText) => clipText)")
+  end
+
+  def click_selection_button(button)
+    selector =
+      case button
+      when "quote"
+        "#chat-quote-btn"
+      when "copy"
+        "#chat-copy-btn"
+      when "cancel"
+        "#chat-cancel-selection-btn"
+      when "move"
+        "#chat-move-to-channel-btn"
+      end
+    within(".chat-selection-management-buttons") { find(selector).click }
+  end
+
+  def copy_messages_to_clipboard(messages)
+    messages = Array.wrap(messages)
+    messages.each { |message| select_message(message) }
+    expect(chat_channel_page).to have_selection_management
+    click_selection_button("copy")
+    expect(page).to have_content("Chat quote copied to clipboard")
+    clip_text = read_clipboard
+    expect(clip_text.chomp).to eq(generate_transcript(messages, current_user))
+    clip_text
+  end
+
+  def generate_transcript(messages, acting_user)
+    messages = Array.wrap(messages)
+    ChatTranscriptService
```

Comment:
I think it makes sense to use this directly rather than typing in the BBCode manually in this test, but open to differing opinions

Reply:
What I mean here is that, without this I would do:

```ruby
expect(page).to have_composer_content(<<~BBCODE)
  [chat quote=\"bruce2;1104;2022-12-14T00:17:44Z\" channel=\"Sports Center 0\" channelId=\"408\"]\nBeep boop\n[/chat]
BBCODE
```

With the values substituted as needed. So I was mostly using `ChatTranscriptService` for convenience. If you think doing the BBCode directly is better I am happy to, just clarifying based on my original question